kernel/term_typing.ml

```diff
@@ -169,6 +169,10 @@ the polymorphic case
   | DefinitionEntry c ->
       let { const_entry_type = typ; const_entry_opaque = opaque ; _ } = c in
       let { const_entry_body = body; const_entry_feedback = feedback_id; _ } = c in
+      let tj = match typ with
+      | None -> None
+      | Some t -> Some (Typeops.infer_type env t)
```

Comment:
Aren't you missing the public universes at that time? And when the type is None the kernel could still infer something using private univs.

Reply:
I think the `let env, usubst, univs, private_univs = match c.const_entry_universes with` block should return 2 environments, and when the type is None run `Typeops.check_wellformed_universes` (needs to be exposed) on the inferred type in the smaller env.

Reply:
You're right, I am refining the patch right now. This piece of code is a mess.

Reply:
I don't know how to ensure cleanly the absence of dependence to private univs in the `None` case without essentially doing the same as #10252 though. Do you have any advice?